mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js

```diff
@@ -0,0 +1,86 @@
+import { useWatch } from 'react-hook-form'
+import PropTypes from 'prop-types'
+import React, { useEffect, useState } from 'react'
+
+import { FormLabel, MenuItem, Stack, TextField } from '@mui/material'
+import { makeValidCamelCaseName } from '../../library/strings/makeValidCamelCaseName'
+import { makeValidClassName } from '../../library/strings/makeValidClassName'
+import { NestedQuestion } from '../../styles/forms'
+import { restorationAims as questions } from '../../data/questions'
+import language from '../../language'
+const stakeholdersPropType = PropTypes.arrayOf(
+  PropTypes.shape({ stakeholderType: PropTypes.string, stakeholderName: PropTypes.string })
+)
+
+const StakeholderBenefitsInputs = ({
+  optionId,
+  parentQuestionFieldName,
+  reactHookFormInstance,
+  stakeholders
+}) => {
+  const questionLabelId = `${optionId}-question-label`
+  const stakeholdersToUse = stakeholders ?? []
+  const {
+    setValue: setFormValue,
+    getValues: getFormValues,
+    control: formControl
+  } = reactHookFormInstance
+  const benefitsFieldName = `${parentQuestionFieldName}StakeholderBenefits`
+  const stakeholderBenefits = useWatch({ control: formControl, name: benefitsFieldName })
+
+  const handleOnChange = ({ event, stakeholderPropertyName }) => {
+    const previousBenefits = getFormValues(benefitsFieldName)
+
+    setFormValue(benefitsFieldName, {
```

Comment:
benefitsFieldName is the parent fieldName. Se we are appending the nested answers to the parent here